### ISSUE TITLE
Enhancements to snippets and syntaxes

### ISSUE THREAD

**Issue by GlenNicholls** [ghdl/ghdl-language-server#32]
The snippets and syntaxes are in rough form right now. The following should be added to both:

## Syntaxes
The following keywords are not currently in the syntaxes

* [ ] rising_edge/falling_edge
* [ ] event
* [ ] all types in ieee: real/integer/signed/unsigned/std_logic/etc. (functions too?)
* [ ] ensure all identifiers are highlighted. Instead of blue, they should be another color

## Snippets
The following need to be added to snippets

* [ ] entity
* [ ] component
* [ ] architecture
* [ ] all types in ieee: real/integer/signed/unsigned/std_logic/etc. Possibly abbreviations for sl/slv etc.
* [ ] package
* [ ] ieee libraries
* [ ] function
* [ ] procedure
* [ ] context (? what other VHDL 2008 features) 
* [ ] process
* [ ] case

**Comment by GlenNicholls** [comment]
@1138-4EB @tgingold feel free to assign this to me and I can start knocking these out as I have time

**Comment by GlenNicholls** [comment]
@tgingold I am going through the current syntaxes to change the highlight colors for all `identifiers`. However, vscode doesn't have an easy way of doing this outside creating a new theme which I think is not good since it will force users to use a certain color scheme. I found this [example](https://github.com/gctse/syntax-highlighting-VS-Code-example/blob/master/log-example/syntaxes/log.tmLanguage.json) from this [blog](https://gcthesoftwareengineer.com/2017/01/how-to-create-custom-syntax-highlighting-in-a-visual-studio-code-extension/)

Basically, it uses a `regexp` to determine which scope is used for highlighting. Does this seem reasonable or is there a better way to highlight non-standard repositories while minimizing the manual intervention to support the default themes of VSC?

**Comment by GlenNicholls** [comment]
@tgingold can I get permissions? I made a branch and I'm starting to commit my changes/additions to the snippets

**Comment by 1138-4EB** [comment]
@GlenNicholls, the regular procedure is to fork this repo to say `GlenNicholls/ghdl-language-server`, using the 'Fork' button above. Then, you push the branch to your repo. Last, you open a Pull Request, by selecting a branch from your repo.

**Comment by GlenNicholls** [comment]
@1138-4EB Can you check out my commit for the snippets in 4ce89e4? I am adding as much as I can and would like to organize the `snippets.vhdl.json` to be a bit more readable. Because comments are not supported, I am grouping everything with delimiters in the description. Do you see value in this or should I forego that and just have a short description for each snippet?

Also, what do you think about the above comment I made about the syntax highlighting?

**Comment by GlenNicholls** [comment]
@1138-4EB @tgingold I am also thinking about adding a snippet for a VHDL file header:

```
"File Header": {
		"prefix": ["FILE_HEADER"],
		"body": [
			"-----------------------------------------------------------------------------",
			"--",
			"-- Title  : ${1:<Title of File>}",
			"-- Design : ${2:<Title of Design>}",
			"-- Author : ${3:<Your Name>}",
			"-- Date   : $CURRENT_MONTH/$CURRENT_DATE/$CURRENT_YEAR $CURRENT_HOUR:$CURRENT_MINUTE",
			"-- File   : <$TM_FILENAME>",
			"--",
			"-----------------------------------------------------------------------------",
			"--",
			"-- Description",
			"--------------",
			"-- ${4:<Enter Description>}",
			"--",
			"-- Interface",
			"------------",
			"--",
			"-- TODO",
			"-------",
			"--",
			"-----------------------------------------------------------------------------"
		],
		"description": "Standard file header"
	},
```

Obviously, the format could change, but I use that all the time. Is there value in adding this to the ghdl-ls vscode plugin, or is that too specific?

**Comment by tgingold** [comment]
Many people want to have an header, but rarely the same.  How easily could it be customized by users ?

**Comment by tgingold** [comment]
I have to investigate.  I think the existing identifiers are already highlighted on my setup.

**Comment by GlenNicholls** [comment]
@tgingold 
>Many people want to have an header, but rarely the same.  How easily could it be customized by users ?
Hmmm, I don't think it would be an easy task. I'm not even sure that it would work with the provided snippet variables. I opened a VSC issue about allowing embedded snippets which would help. At this time, though, I don't think it would be possible for a user to customize the header without a macro or something.
>I have to investigate.  I think the existing identifiers are already highlighted on my setup.
Okay let me know. I have tried all default themes and turned off all VHDL extensions except ghdl-ls and they aren't highlighted at all for me.

**Comment by tgingold** [comment]
Here is what I get:
![Screenshot from 2019-11-14 14-09-57](https://user-images.githubusercontent.com/13433645/68882687-f84e3400-070f-11ea-90e2-9506c5a6ce57.png)

**Comment by GlenNicholls** [comment]
@tgingold 

> Here is what I get:
> ![Screenshot from 2019-11-14 14-09-57](https://user-images.githubusercontent.com/13433645/68882687-f84e3400-070f-11ea-90e2-9506c5a6ce57.png)

Do you have an `hdl-proj.json` file. In the default theme, I don't see any highlighting for the `identifiers` at all :/
